flake.nix

```diff
@@ -534,15 +590,18 @@
                       wantedBy = [ "multi-user.target" ];
                     };
 
-                  # firewall
-                  networking.firewall = {
-                    allowedTCPPorts = [ 9000 ];
-                    allowedUDPPorts = [ 9000 ];
-                    interfaces."wg0".allowedTCPPorts = [
-                      5052 # TODO: use 'lighthouse.endpoint.port' here by converting it to u16
-                    ];
-                  };
-                })
+                    # firewall
+                    networking.firewall = {
+                      allowedTCPPorts = [ 9000 ];
+                      allowedUDPPorts = [ 9000 ];
+                      interfaces = lib.mkIf (cfg.${activeVPNClient}.enable) {
+                        "${cfg.${activeVPNClient}.interfaceName}".allowedTCPPorts = [
+                          (lib.strings.toInt endpoint.port)
```

Comment:
The `endpoint.port` is a bit vague: I see it's defined as lighthouse end point in the let header, but having it be called in this context makes it easy to misthink that the endpoint.port could be related to the VPN client, not CL. It might be a better idea to have the port parsing be done in the config block, or somehow joining the data structure together.

Reply:
Just renamed for now. I think the parsing should be done via a function, which could have a place at the top of the module configuration alongside other functions. It seemed a bit troublesome so I didn't bother to try creating it right now. It might be easier to just revert back to defining them separately from the get-go after all. This should work for now tho.